test/create-user-test.ts

```diff
@@ -0,0 +1,272 @@
+        },
+      }))!;
+      expect(userCreated.name).to.be.deep.eq(inputCreateUser.name);
+      expect(userCreated.email).to.be.deep.eq(inputCreateUser.email);
+      expect(userCreated.birthDate).to.be.deep.eq(inputCreateUser.birthDate);
+      const verifyHash = await verifyPassword(userCreated.password, inputCreateUser.password);
+      expect(verifyHash).to.be.eq(true);
+      expect(response.status).to.be.deep.eq(201);
+      expect(response.data).to.be.deep.eq({
+        id: userCreated.id,
+        name: inputCreateUser.name,
+        email: inputCreateUser.email,
+        birthDate: inputCreateUser.birthDate,
+      });
+    });
+
+    describe('Errors creating user', () => {
+      it('Error when jwt is invalid', async () => {
+        const errorJwt = {
+          code: 'USR_10',
+          message: 'authentication failure',
+          details: 'Invalid Token',
+        };
+        try {
+          await endpoint.post('/users', inputCreateUser, {
+            headers: {
+              Authorization: '12344jajaja',
+            },
+          });
+        } catch (error: any) {
+          expect(error.response.status).to.be.deep.eq(401);
+          expect(error.response.data).to.be.deep.eq(errorJwt);
+        }
+      });
+    });
+
+    it('Error when email already exists', async () => {
+      const input: UserCreateReq = {
+        name: 'Gabrielle',
+        email: 'gabi@gmail.com',
+        password: 'coxinha123',
+        birthDate: '02-12-2000',
+      };
+      try {
+        await endpoint.post('/users', input, {
+          headers: {
+            Authorization: token,
+          },
+        });
+      } catch (error: any) {
+        expect(error.response.status).to.be.deep.eq(400);
+        expect(error.response.data).to.be.deep.eq({
+          code: 'USR_03',
+          message: 'Invalid credential',
+          details: 'There is already a user registered with this email. Please provide a different email address.',
+        });
+      }
+    });
+
+    describe('Password invalid', () => {
+      const inputPasswordWithoutDigit: UserCreateReq = {
+        name: 'Gabrielle',
+        email: 'gabi@gmail.com',
+        password: 'coxinha',
+        birthDate: '02-12-2000',
+      };
+      it('Error when password does not contain a digit', async () => {
+        try {
+          await endpoint.post('/users', inputPasswordWithoutDigit, {
+            headers: {
+              Authorization: token,
+            },
+          });
+        } catch (error: any) {
+          expect(error.response.status).to.be.deep.eq(400);
+          expect(error.response.data).to.be.deep.eq({
+            code: 'USR_01',
+            message: 'Password must contain at least one digit.',
+            details:
+              'The password must contain at least 6 characters and among them there must be at least 1 letter and 1 digit.',
+          });
+        }
+      });
+    });
+    it('Error when password is less than 6 characters', async () => {
```

Comment:
Nit: recomendo deixar uma linha em branco entre cada "it" pra ajudar na leitura.